packages/apollo-engine-reporting/src/agent.ts

```diff
@@ -196,7 +212,59 @@ export class EngineReportingAgent<TContext = any> {
     );
   }
 
-  public addTrace(signature: string, operationName: string, trace: Trace) {
+  private async getTraceSignature({
+    queryHash,
+    operationName,
+    documentAST,
+    queryString,
+  }: {
+    queryHash: string;
+    operationName: string;
+    documentAST?: DocumentNode;
```

Comment:
If this is always either a document AST or a string, should this be a single `DocumentNode | string` argument?